addon/index.ts

```diff
@@ -0,0 +1,3 @@
+export { default, InTeardown } from "./-private/class/modifier";
```

Comment:
Is `InTeardown` externally visible? If so, how?

Reply:
It is, exported as a type utility so that users can do this to work around the limitations of the way the current API interacts with the type system if so desired:

```ts
import Modifier, { InTeardown } from 'ember-modifier';

export default class UsesWillDestroy {
  willDestroy(this: InTeardown<UsesWillDestroy>) {
    this.element // null
  }
}
```

Reply:
Ummm, yuck 😝 

Seriously though, we should strongly consider removing the nulling of `this.element`. I think in practical terms it is not doing us any favors and it is extremely unlikely to be a reasonable stop gap for memory leaks (which is likely the reason it was added).

Reply:
I strongly agree, and will be opening an issue with proposed changes to the API *after* we land this in a stable release. (🙅 to refactors that also introduce changes, as I know you feel as well.)

Reply:
Tracking *this* follow-up at #29.